test/e2e/RunScenario.spec.ts

```diff
@@ -0,0 +1,217 @@
+import app from '../../src/server';
+import DB from '../../src/models';
+import express = require('express');
+import { Router } from 'express';
+import * as utils from "../utils/utils";
+
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+
+chai.use(chaiHttp);
+const {expect} = chai;
+
+const APIKEY = '7718330d2794406c980bdbded6c9dc1d';
+const source = `
+          #include <iostream>
+          using namespace std;
+          int main () {
+              char in[20];
+              cin>>in;
+              cout<<in;
+              return 0;
+          }`;
+const stdin = 'Success';
+const expectedOutput = 'Success';
+
+
+function delay(ms: number) {
+    return new Promise( resolve => setTimeout(resolve, ms) );
+}
+
+describe('POST api/runs', () => {
+    before(async () => {
+        await DB.apikeys.create({
+            id: 1,
+            key: APIKEY,
+            whitelist_domains: ['*'],
+            whitelist_ips: ['*']
+        });
+        await DB.langs.create({
+            lang_slug: 'cpp',
+            lang_name: 'C++',
+            lang_version: '11'
+        });
+        await utils.setupMockQueue()
+    });
+    after(utils.truncateTables);
+
+
+    it('should throw 403 error if API Key is absent', async () => {
+        // shouldn't this be bad request 400?
```

Comment:
400 represents Bad request, like unwanted keys in the body. Whereas Unauthorized is sent if we are missing authorization information.

Clients will straightaway know that they are missing Authorization header if they receive 403